op-node/rollup/derive/system_config.go

```diff
@@ -75,33 +78,41 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 	// Allocate a placeholder word to read into
 	word := make([]byte, 32)
 
+	// Helper function to prevent code duplication.
+	readIntoWord := func(b []byte) {
+		// Sanity check that the byte array we're reading into is always 32 bytes in length.
+		if len(b) != 32 {
+			panic("invalid word length")
+		}
+
+		if _, err := reader.Read(b); err != nil {
+			// The possible error returned by `Read` is ignored due to the length check of the unindexed data in
+			// all cases of the below switch statement. While we don't panic here, this log should *never* be emitted.
+			logger.Crit("failed to read word from unindexed log data")
+		}
+	}
+
 	// Attempt to read unindexed data
 	switch updateType {
 	case SystemConfigUpdateBatcher:
 		if len(ev.Data) != 32*3 {
 			return fmt.Errorf("expected 32*3 bytes in batcher hash update, but got %d bytes", len(ev.Data))
 		}
 
-		// Attempt to read the pointer, it should always equal 32.
-		if _, err := reader.Read(word); err != nil {
-			return fmt.Errorf("failed to read pointer in batcher address update")
-		}
+		// Read the pointer, it should always equal 32.
+		readIntoWord(word)
```

Comment:
Nit: maybe `word := readWord()` works better, to avoid repeatedly using the same word? Can do `if word := readWord(); word == ...`  inline to avoid definitions colliding.

Reply:
Sure! I like how this looks more. I originally did it this way so that go wouldn't have to allocate new memory each time a word was read, but still getting familiar with the quirks of the compiler. Would this involve re-allocations?

Reply:
If it returns it should just allocate on the stack if you use small fixed-size arrays and local variables only.  That said, I think readability is more important than performance in this codepath.